### PR TITLE
Paramedic role, redone

### PR DESCRIPTION
**Pull request by MagnusCrowe**
## About the PR

This PR adds the paramedic role officially to the game.  This is an updated version of https://github.com/space-wizards/space-station-14/pull/14055.

- Paramedic has access to medical, external, and maintenance.
- Added a brand new locker stocked with their void suit, clothing, and a medical HUD. _Expect  another PR later that removes the void suit and helmet from the medidrobe, once all the maps are updated to include this locker._
- Paramedics have an experimental PDA that has a faster scan rate than other doctors, scanning a patient in half a second. Perfect for quick field triage.
- Paramedics will start with a medical belt that has an inventory slightly more geared for their profession.
- The role will require 15 hours of general gameplay and at least 2.5 hours as a medical doctor.  They are expected to know how to stabilize and transport a patient.  They are also expected to know how to handle corpses to prevent their degradation. 

The role is expected to spend little time in med bay, instead spending time out roaming the station or responding to emergencies.  They are expected to stabilize patients and transport them to med bay for the medical doctors to treat fully.

**Media**

ParamedicDisplay2

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added the paramedic role

### CODE REVIEW

**Comment by PuroSlavKing** [comment]
Team Fortress 2 role icon

**Comment by metalgearsloth** [comment]
This seems significantly stronger than an MD?

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml [hunk 1]
```diff
@@ -89,3 +89,18 @@
...
+  components:
+  - type: StorageFill
+    contents:
+      - id: ClothingHeadHelmetVoidParamed
+      - id: ClothingOuterHardsuitVoidParamed
+      - id: ClothingOuterCoatParamedicWB
+      - id: ClothingHeadHatParamedicsoft
+      - id: ClothingUniformJumpsuitParamedic
+      - id: ClothingUniformJumpskirtParamedic
+      - id: ClothingEyesHudMedical
```
Comment (PuroSlavKing) [inline Resources/Prototypes/Catalog/Fills/Lockers/medical.yml:106]: make line 107 empty

**Comment by Emisse** [comment]
they dont need service access

**Comment by Emisse** [comment]
i'd rather pda med scanner be converted to cartridge rather than an intrinsic component

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml (hunk 1 above)
Comment (MagnusCrowe) [reply Resources/Prototypes/Catalog/Fills/Lockers/medical.yml]: Done

**Comment by MagnusCrowe** [comment]
> they dont need service access

Removed

**Comment by MagnusCrowe** [comment]
> i'd rather pda med scanner be converted to cartridge rather than an intrinsic component

That sounds fine to me, but beyond the scope of this PR.

**Comment by Myakot** [comment]
I'm no code maintainer or balance freak, but I really dislike the fact that instead of actually being a paramedic, the only thing people on this role will do is:
- Find someone
- Shove every single med into the mouth of victim
- Victim wakes up, 0 dmg, ready to continue on, without even visiting medbay.

I honestly think it'd be much better if Paramedic had an insanely good stretchers (for an example - stasis stretchers) which would actually allow Paramedic to be a Paramedic and not a mobile medbay. Why would he have a better PDA compared to CMO?

**Review by QuietlyWhisper — APPROVED** [review]
Whatever maintainer looks at this, the key things you may want to look at are the belt (they have 6 medpens), the crew monitor in their pocket, and faster scanning speed on their PDA.

**Comment by Emisse** [comment]
I like paramedic I am just hesitant on their role before the body refactor is out, that said I'm not opposed to merging now. I approve but I'll wait for more opinions.

**Review by Emisse — APPROVED** [review]

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+- type: startingGear
+  id: ParamedicGear
+  equipment:
+    jumpsuit: ClothingUniformJumpsuitParamedic
+    back: ClothingBackpackMedicalFilled
+    shoes: ClothingShoesColorWhite
+    id: ParamedicPDA
+    ears: ClothingHeadsetMedical
+    pocket1: HandheldCrewMonitor
```
Comment (metalgearsloth) [inline Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml:31]: This is a lategame tech, please don't add it to them.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
This deviates a bit much from being a flavour role for MD.

It's similar to warden and AT except in those cases the separation is required due to the amount of damage they can do to the station, except for paramedic it's just an MD that leaves their department.

**Comment by deltanedas** [comment]
syndie parameds are a complete nightmare so this still applies

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    - !type:OverallPlaytimeRequirement
+      time: 54000
+  startingGear: ParamedicGear
+  icon: "Paramedic"
+  supervisors: job-supervisors-cmo
+  access:
+  - Medical
+  - Maintenance
+  - External
+  - Paramedic
```
[the remark this answers is not in the document]
Comment (QuietlyWhisper) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml:19]: See the locker that was added

**Comment by QuietlyWhisper** [comment]
> This deviates a bit much from being a flavour role for MD.
> 
> It's similar to warden and AT except in those cases the separation is required due to the amount of damage they can do to the station, except for paramedic it's just an MD that leaves their department.
 
Paramedics are designed for the express purpose of leaving medical and recovering lost bodies while doctors stay in med bay, it's not flavor role here actually

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml:19]: I saw it, it shouldn't be its own access.

[the remark this answers is not in the document]
Comment (MagnusCrowe) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml:19]: Another possibility may be allowing lockers to have multiple access requirements. Let's say you need medical AND external to open the paramedic locker.  Could also be used for brigmedic, security and medical to open.

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml (hunk 2 above)
Comment (MagnusCrowe) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml:31]: Should we remove it from laying around on most stations and from the CMO too?  Serious question, not sarcasm.

**Comment by MagnusCrowe** [comment]
> I'm no code maintainer or balance freak, but I really dislike the fact that instead of actually being a paramedic, the only thing people on this role will do is:
> 
> * Find someone
> * Shove every single med into the mouth of victim
> * Victim wakes up, 0 dmg, ready to continue on, without even visiting medbay.
> 
> I honestly think it'd be much better if Paramedic had an insanely good stretchers (for an example - stasis stretchers)

I love that idea, mind if I steal it?

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (MagnusCrowe) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml]: Paramedic access removed altogether.

File: Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml (hunk 2 above)
Comment (MagnusCrowe) [reply Resources/Prototypes/Roles/Jobs/Medical/paramedic.yml]: Removed.

**Comment by MagnusCrowe** [comment]
I made the request changes 2 weeks ago @metalgearsloth

**Pull request merged**